google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/PublisherImpl.java

```diff
@@ -209,12 +210,12 @@ private void processBatch(Collection<UnbatchedMessage> batch) throws StatusExcep
       return ApiFutures.immediateFailedFuture(error.asException());
     }
     try (CloseableMonitor.Hold h = monitor.enter()) {
-      if (shutdown) {
-        return ApiFutures.immediateFailedFuture(
-            Status.FAILED_PRECONDITION
-                .withDescription("Published after the stream shut down.")
-                .asException());
-      }
+      ApiService.State currentState = state();
+      checkState(
+          currentState == ApiService.State.RUNNING,
```

Comment:
checkState(state() == ApiService.State.RUNNING, ...);

Reply:
The state is also used for the error message. I saw that state() does some enum mappings that we could avoid doing twice:
https://github.com/googleapis/api-common-java/blob/master/src/main/java/com/google/api/core/AbstractApiService.java#L125

Reply:
SGTM.